src/app/pages/home/home.component.spec.ts

```diff
@@ -1,9 +1,10 @@
 import { ProductsComponent } from './../../components/products/products.component';
 import { HomeComponent } from './home.component';
 import { MockBuilder, MockRender } from 'ng-mocks';
+import {AppModule} from "../../app.module";
 describe('Home Component', () => {
   beforeEach(() => {
-    return MockBuilder(HomeComponent).mock(ProductsComponent)
+    return MockBuilder(HomeComponent, AppModule)
```

Comment:
instead of providing dependencies in tests, it's better to specify the related module.

Reply:
Will it be a "throw all under the kitchen sink approach" or specifying a module which contains all other pieces, or is it better to just put a mock for a respective service.. will there be any difference when it comes to test suites getting slower or something? Just asking to understand.

Reply:
Ideally you should have here `HomeModule` which contains only dependencies for `HomeComponent`.

When you explicitly provide `ProductsComponent` in tests, there won't be a failure if someone deletes `ProductsComponent` from `AppModule`.

When you provide a component and its module only, tests will fail if any of required dependencies is missing.

Reply:
got it! thank you, so ideally, each "page" or each route if done as a separate module, it increases the granularity . For this small app, i believe, I only had an app module and shoved everything into one module. Will refactor the application slightly